staging/src/k8s.io/apiserver/pkg/server/options/tracing.go

```diff
@@ -19,24 +19,37 @@ package options
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
 	"net"
 
 	"github.com/spf13/pflag"
 	"go.opentelemetry.io/otel/exporters/otlp/otlpgrpc"
 	"go.opentelemetry.io/otel/sdk/resource"
-	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/semconv"
 	"google.golang.org/grpc"
 
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apiserver/pkg/apis/apiserver"
+	"k8s.io/apiserver/pkg/apis/apiserver/install"
 	"k8s.io/apiserver/pkg/server"
 	"k8s.io/apiserver/pkg/server/egressselector"
-	"k8s.io/apiserver/pkg/tracing"
+	v1alpha1componentbaseconfig "k8s.io/component-base/config/v1alpha1"
```

Comment:
Since the old types aren't going away, do we need to do this now or can we save the cleanup for later?

Reply:
this PR updates the handling of TracingConfiguration to the component-base, but accepts the apiserver.TracingConfiguration read in from the config file. As soon as the internal tracing configuration is decoded from the config file, it is converted into the new component-base/config struct.